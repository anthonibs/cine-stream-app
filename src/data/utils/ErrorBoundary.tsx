import { Component, ErrorInfo, ReactNode } from 'react';

interface ErrorBoundaryProps {
	children: ReactNode;
}

interface ErrorBoundaryState {
	hasError: boolean;
}

class ErrorBoundary extends Component<ErrorBoundaryProps, ErrorBoundaryState> {
	constructor(props: ErrorBoundaryProps) {
		super(props);
		this.state = { hasError: false };
	}

	handleReload = () => window.location.reload();

	handleBack = () => window.location.replace('/');

	componentDidCatch(error: Error, errorInfo: ErrorInfo): void {
		this.setState({ hasError: true });
		console.error('Erro encontrado: ', error, errorInfo);
	}

	render(): ReactNode {
		if (this.state.hasError) {
			return (
				<div>
					<h2>
						Oops! Um erro inesperado aconteceu. Por favor recarregue a página ou volte ao inicio.
					</h2>
					<img src='/not_found.webp' alt='sem resultado' />
					<button onClick={this.handleReload}>
						<h3>Recarregar Página</h3>
					</button>
					<button onClick={this.handleBack}>
						<h3>Voltar ao Inicio</h3>
					</button>
				</div>
			);
		}

		return this.props.children;
	}
}

export default ErrorBoundary;
