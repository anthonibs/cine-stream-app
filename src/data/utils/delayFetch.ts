export default function delayFetch(ms = 1000) {
	return new Promise((resolve) => setTimeout(resolve, ms));
}
