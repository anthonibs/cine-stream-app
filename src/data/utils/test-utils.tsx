import { ReactElement } from 'react';
import { BrowserRouter } from 'react-router-dom';

import { RenderOptions, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import { AuthProvider } from 'data/contexts/Auth';
import { LanguageProvider } from 'data/contexts/Language';
import { MyFavoritesProvider } from 'data/contexts/MyFavoritesList';

import { ThemeProvider } from 'styled-components';

import GlobalStyles from 'ui/styles/GlobalStyles';
import dark from 'ui/styles/themes/dark';

interface ProvidersProps {
	children: ReactElement;
}

const Providers = ({ children }: ProvidersProps) => {
	return (
		<ThemeProvider theme={dark}>
			<GlobalStyles />
			<AuthProvider>
				<LanguageProvider>
					<MyFavoritesProvider>
						<BrowserRouter>{children}</BrowserRouter>
					</MyFavoritesProvider>
				</LanguageProvider>
			</AuthProvider>
		</ThemeProvider>
	);
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
	render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';
export { customRender as render };
