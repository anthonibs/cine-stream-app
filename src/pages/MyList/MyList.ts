import styled, { keyframes } from 'styled-components';

const fadeInSerie = keyframes`
	to {
		opacity: 1;
	}
	from{
		opacity: 0;
	}
`;

const fadeInMovie = keyframes`
	to {
		opacity: 0;
	}
	from{
		opacity: 1;
	}
`;


export const StyledContainer = styled.div`
	height: 100%;
	min-height: calc(100vh - 317px);
	padding-top: 90px;
	width: 100%;
`;


export const StyledSectionMyFavorites = styled.section`
	@media (min-width: 375px) {
		margin: 1.5rem auto 5rem;
		width: 85%;
	}

	@media (min-width: 768px) {
		margin: 2rem auto 5rem;
		width: 90%;
	}

	@media (min-width: 1024px) {
		width: 95%;
	}
`;


export const StyledGrid = styled.div`
	display: grid;
	position: relative;
	width: 100%;
	opacity: 1;

	&.serie {
		animation: ${fadeInSerie} 1s;
	}

	&.movie {
		animation: ${fadeInMovie} 1s alternate-reverse;
	}

	@media (min-width: 375px) {
		gap: 3rem .6rem;
		grid-template-columns: repeat(auto-fill, minmax(140px, 1fr));
		margin-top: 1.6rem;
	}

	@media (min-width: 600px) {
		gap: 3rem 1rem;
	}

	@media (min-width: 968px) {
		grid-template-columns: repeat(auto-fill, minmax(180px, 1fr));
	}
`;


export const StyledHeaderColumn = styled.div`
	display: flex;
	width: 100%;
	justify-content: space-between;
`;


export const StyledMessageContainer = styled.span`
	text-align: center;
	display: block;
	margin-top: 4rem;
	pointer-events: none;
	user-select: none;
	height: 100%;

	&.serie {
			animation: ${fadeInSerie} 2s;
		}

	&.movie {
		animation: ${fadeInMovie} 2s alternate-reverse;
	}

	& > svg {
		font-size: 6rem;
		color: ${({ theme }) => theme.font.color[100]};
		margin-bottom: 1rem;
	}

	& > span {
		display: block;
		margin-top: .325rem;
		color:  ${({ theme }) => theme.font.color[100]};
	}
`;