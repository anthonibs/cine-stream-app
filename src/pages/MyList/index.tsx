import { useMemo, useState } from 'react';

import {
	StyledContainer,
	StyledGrid,
	StyledSectionMyFavorites,
	StyledHeaderColumn,
} from './MyList';

import { useMyFavoritesList } from 'data/hooks/useMyFavoritesList';
import useLanguage from 'data/hooks/useLanguage';

import CardPoster from 'ui/components/common/CardPoster';
import CardPosterSerie from 'ui/components/common/CardPosterSerie';
import Select from 'ui/components/common/Select';
import Heading from 'ui/components/common/Typography/Heading';

import options from './translate.json';


const MyList = () => {

	const { language } = useLanguage();
	const { listMovie, listSerie } = useMyFavoritesList();

	const [selectedListType, setSelectedListType] = useState('movie');

	const selectedList = useMemo(() => {
		if (selectedListType === 'movie') {
			return listMovie.map(item => <CardPoster key={item.id} poster={item} />);
		}
		if (selectedListType === 'serie') {
			return listSerie.map(item => <CardPosterSerie key={item.id} poster={item} />);
		}

		if (selectedListType === '') {
			return listMovie.map(item => <CardPoster key={item.id} poster={item} />);
		}
	}, [listMovie, listSerie, selectedListType]);


	const translate = useMemo(() => {
		return options.option_favorites.find(item => item.code === language);
	}, [language]);

	return (
		<StyledContainer>
			<StyledSectionMyFavorites>
				<StyledHeaderColumn>
					<Heading component='h1' variant='h4'>
						{translate?.title}
					</Heading>
					<Select
						state={translate?.options}
						setState={setSelectedListType}
						defaultValue={translate?.options[0].name}
						position='absolute'
					/>
				</StyledHeaderColumn>
				<StyledGrid
					className={selectedListType}
				>
					{selectedList}
				</StyledGrid>
			</StyledSectionMyFavorites>
		</StyledContainer>
	);
};

export default MyList;