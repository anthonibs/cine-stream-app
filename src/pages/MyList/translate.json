{
	"option_favorites": [
		{
			"code": "pt-BR",
			"options": [
				{
					"id": "movie",
					"name": "Filmes"
				},
				{
					"id": "serie",
					"name": "SÃ©ries"
				}
			],
			"title": "Minha lista"
		},
		{
			"code": "en-US",
			"options": [
				{
					"id": "movie",
					"name": "Movies"
				},
				{
					"id": "serie",
					"name": "Series"
				}
			],
			"title": "My list"
		}
	]
}