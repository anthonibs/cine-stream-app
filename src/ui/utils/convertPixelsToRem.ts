const pixelsDefault = 16;

export default function convertPixelsToREM(pixel: number): string {
	return `${pixel / pixelsDefault}rem`;
}