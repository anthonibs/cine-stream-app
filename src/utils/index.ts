export * from './convertMinutesToHours';
export * from './combinedListFavorites';
export * from './convertPixelsToRem';
export * from './removeAccentsFromText';
export * from './differenceBetweenDates';
