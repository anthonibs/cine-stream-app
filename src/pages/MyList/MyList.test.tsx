import { fireEvent, render, screen } from 'data/utils/test-utils';
import MyList from '.';

describe('Must render <MyList/>', () => {
	it('Must render component MyList', () => {
		render(<MyList />);

		const snapshotMyList = screen.getByTestId('mylist-styles');
		expect(snapshotMyList).toMatchSnapshot('Styles');
	});

	it('Must render page heading level one', () => {
		render(<MyList />);

		const title = screen.getByRole('heading', { level: 1 });
		expect(title).toBeInTheDocument();
	});

	it('Must render the title in pt-BR', () => {
		render(<MyList />);

		const myList = screen.getByText('Minha lista');
		expect(myList).toBeInTheDocument();
	});

	it('Must call list select button', () => {
		render(<MyList />);
		const button = screen.getByTestId('selected-mylist');
		const list = screen.getByTestId('selected');

		expect(button).toBeInTheDocument();
		expect(list).toBeInTheDocument();

		expect(list).not.toBeVisible();

		fireEvent.click(button);

		expect(list).toHaveAttribute('aria-hidden', 'false');
		expect(list).toBeVisible();
	});
});
