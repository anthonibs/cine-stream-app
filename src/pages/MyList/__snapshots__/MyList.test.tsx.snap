// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Must render <MyList/> Must render component MyList: Styles 1`] = `
<div
  class="sc-bgqQcB cPkYiK"
  data-testid="mylist-styles"
>
  <section
    class="sc-gTRrQi eFTMUq"
  >
    <header
      class="sc-fFGjHI cHGHNs"
    >
      <div>
        <h1
          class="sc-ejWHKt hiecth"
          color="secondary"
        >
          Minha lista
        </h1>
        <div
          class="sc-fWFeAW gLcOIS"
        >
          <button
            aria-labelledby=""
            class="sc-gWHAAX exexyl"
            data-testid="selected-mylist"
            type="button"
          >
            Filmes
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
              />
            </svg>
          </button>
          <div
            aria-hidden="true"
            class="sc-ccVCaX hvgRif"
            data-testid="selected"
            role="select"
          >
            <span
              class="sc-jnqLxu geRDYA"
              id="Filmes"
              role="option"
            >
              Filmes
            </span>
            <span
              class="sc-jnqLxu geRDYA"
              id="Séries"
              role="option"
            >
              Séries
            </span>
          </div>
        </div>
      </div>
    </header>
    <span
      class="sc-dicizt kFRtkK movie"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M448 256c0-106-86-192-192-192S64 150 64 256s86 192 192 192 192-86 192-192z"
          fill="none"
          stroke-miterlimit="10"
          stroke-width="32"
        />
        <path
          d="M256 176v160m80-80H176"
          fill="none"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="32"
        />
      </svg>
      <p
        class="sc-eWhHU kxHmxX"
      >
        Não há nada na sua lista.
      </p>
      <span>
        O conteúdo que você colocar na Minha Lista aparecerá aqui.
      </span>
    </span>
  </section>
</div>
`;
